packages/apputils-extension/src/index.ts

```diff
@@ -335,24 +335,31 @@ export const toggleHeader: JupyterFrontEndPlugin<void> = {
 async function updateTabTitle(workspace: string, db: IStateDB, name: string) {
   const data: any = await db.toJSON();
   let current: string = data['layout-restorer:data']?.main?.current;
-  if (current === undefined) {
+  // File name from current path
+  let currentFile: string = PathExt.basename(current)?.split(':')[0];
```

Comment:
Thanks! It is not obvious to me what the split on `:` does. Could you clarify, please?

I'm also thinking that it would be a good idea to add unit tests case for the entire `updateTabTitle` function, but it could be done in a follow up PR together with the full user-customization (in short the logic could probably be moved to a new function in `apputils`, say `createTitle(template, workspace, db, name): string`, while `apputils-extension` would only take care of setting the title).

Reply:
The PathExt.basename returns the file names with their extensions appended to it. For example, package.json:JSON. The previous code also already had this logic of using `.split(':')` so I continued it here. Otherwise, filenames with less than 15 characters will also be truncated due to the appended `:extension`. 

About the unit test case, I think that will be helpful for this function. It can definitely be added along with the user-customizable functionality in a single PR as you mentioned.

Reply:
Makes sense! I mean, I am not a fan of how the private data structure of layout restorer is used here, but I guess we can just stick with that for this PR. I added a new suggestion on this below.

Reply:
I've checked the suggestion and committed accordingly.